src/main/java/itm/oss/splitter/Settlement.java

```diff
@@ -1,12 +1,69 @@
 package itm.oss.splitter;
 
+import java.math.BigDecimal;
 import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
 
+/**
+ * Settlement - Suggests payments to clear all debts.
+ * Uses a simple greedy algorithm.
+ */
 public class Settlement {
 
   public static ArrayList<Payment> suggest(Balance b) {
-    // TODO (Issue 5): greedy match largest debtor with largest creditor until
-    // settled.
-    throw new UnsupportedOperationException("suggest() not implemented yet");
+    ArrayList<Payment> pay = new ArrayList<>();
+
+    // Get all balances (name -> amount)
+    Map<String, BigDecimal> map = b.asMap();
+
+    // Split into debtors (negative) and creditors (positive)
+    List<Map.Entry<String, BigDecimal>> neg = new ArrayList<>();
+    List<Map.Entry<String, BigDecimal>> pos = new ArrayList<>();
+    BigDecimal eps = new BigDecimal("0.005");
+
+    for (Map.Entry<String, BigDecimal> e : map.entrySet()) {
+      BigDecimal v = e.getValue();
+      if (v == null)
+        continue;
+      if (v.compareTo(BigDecimal.ZERO) < 0 && v.abs().compareTo(eps) > 0)
+        neg.add(e);
+      else if (v.compareTo(BigDecimal.ZERO) > 0 && v.abs().compareTo(eps) > 0)
+        pos.add(e);
+    }
```

Comment:
The logic inside this loop is critical for filtering valid balances and ignoring tiny amounts based on `eps`. Adding a few comments would greatly improve the clarity more for future maintainers.

I suggest adding comments to explain:
1. The purpose of skipping `null` values.
2. The role of `eps` in filtering balances before adding them to `neg` or `pos` lists.

I suggest commenting like:
```suggestion
        for (Map.Entry<String, BigDecimal> e : map.entrySet()) {
            BigDecimal v = e.getValue();
            if (v == null) continue; // Skip entries with null balance (should not happen if data is clean)
            
            // Only consider negative balances larger than epsilon (0.005) for settlement
            if (v.compareTo(BigDecimal.ZERO) < 0 && v.abs().compareTo(eps) > 0) 
                neg.add(e);
            // Only consider positive balances larger than epsilon (0.005) for settlement
            else if (v.compareTo(BigDecimal.ZERO) > 0 && v.abs().compareTo(eps) > 0) 
                pos.add(e);
        }

Reply:
Thank you for the detailed feedback! I agree that adding comments for these parts will make the code easier to understand. I’ll update it accordingly.

Reply:
Perfect!! It's much clearer now and easy to follow. Thanks for implementing the change.